src/bmi_cfe.c

```diff
@@ -1922,6 +1995,31 @@ static int Set_value_at_indices (Bmi *self, const char *name, int * inds, int le
     if (status == BMI_FAILURE)
         return BMI_FAILURE;
     memcpy(ptr, src, var_item_size * len);
+    
+    if (strcmp (name, "maxsmc") == 0 || strcmp (name, "alpha_fc") == 0 || strcmp (name, "wltsmc") == 0 || strcmp (name, "maxsmc") == 0 || strcmp (name, "b") == 0 || strcmp (name, "slope") == 0 || strcmp (name, "satpsi") == 0 || strcmp (name, "Klf") == 0  || strcmp (name, "satdk") == 0){
```

Comment:
Ok, perhaps my previous comments were immature as I look at this block of code.  Although, I will point out the code duplication between the `init` function and this if block, and recommend at some point refactoring the common blocks into functions, such as
```C
cfe_ptr->direct_runoff_params_struct.Schaake_adjusted_magic_constant_by_soil_type = cfe_ptr->NWM_soil_params.refkdt * cfe_ptr->NWM_soil_params.satdk / 0.000002;
```
could be

```C
cfe_ptr->direct_runoff_params_struct.Schaake_adjusted_magic_constant_by_soil_type = compute_magic_constant(&NWM_soil_params)
```
with
```C
static inline double compute_magic_constant(soil_params* params){
return params->refkdt * params->satdk / 0.000002;
}
```
Just as good practice, especially when variables are computed with additional "magic constants", e.g. 0.000002